### PR TITLE
New unique: May purchase other civs tiles

### PR DESCRIPTION
**Pull request by Dimensiondog**
Draft as I still haven't figured out how to build unciv to test and also unsure how to make sure a player cannot buy their own tiles.

### CODE REVIEW

**Comment by yairm210** [comment]
This seems ridiculously overpowered

File: core/src/com/unciv/logic/city/managers/CityExpansionManager.kt [hunk 1]
```diff
@@ -58,7 +58,13 @@ class CityExpansionManager : IsPartOfGameInfoSerialization {
     fun canBuyTile(tile: Tile): Boolean {
         return when {
             city.isPuppet || city.isBeingRazed -> false
-            tile.getOwner() != null -> false
+            tile.getOwner() != null -> {
```
Comment (yairm210) [inline core/src/com/unciv/logic/city/managers/CityExpansionManager.kt:61]: If X true else false is exactly like just saying "X"

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt [hunk 2]
```diff
@@ -136,6 +136,8 @@ enum class UniqueType(val text: String, vararg targets: UniqueTarget, val flags:
     /// Border growth
     BorderGrowthPercentage("[relativeAmount]% Culture cost of natural border growth [cityFilter]", UniqueTarget.Global, UniqueTarget.FollowerBelief),
     TileCostPercentage("[relativeAmount]% Gold cost of acquiring tiles [cityFilter]", UniqueTarget.FollowerBelief, UniqueTarget.Global),
+    CanPurchaseOtherCivTiles("May purchase other Civs Tiles", UniqueTarget.FollowerBelief, UniqueTarget.Global),
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/unique/UniqueType.kt:139]: As a *follower belief* means that this needs to take the city unique into account, which isn't what's happening now

Comment (Dimensiondog) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt:139]: > As a _follower belief_ means that this needs to take the city unique into account, which isn't what's happening now

Isn't the check for if a city can purchase a tile city based?

File: core/src/com/unciv/logic/city/managers/CityExpansionManager.kt (hunk 1 above)
Comment (Dimensiondog) [reply core/src/com/unciv/logic/city/managers/CityExpansionManager.kt:61]: Im confused by this

Comment (yairm210) [reply core/src/com/unciv/logic/city/managers/CityExpansionManager.kt:61]: "If (<condition>) true else false" is exactly the same as just writing the condition

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt (hunk 2 above)
Comment (yairm210) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt:139]: The uncivInfo.hasUnique you wrote should check if the *city* has the unique

**Comment by Dimensiondog** [comment]
Changed the code hope it works better now.

**Comment by alexban011** [comment]
```
e: file:///home/runner/work/Unciv/Unciv/core/src/com/unciv/logic/city/managers/CityExpansionManager.kt:61:32 Unresolved reference: civ
e: file:///home/runner/work/Unciv/Unciv/core/src/com/unciv/logic/city/managers/CityExpansionManager.kt:61:40 Unresolved reference: uncivInfo
```

Well for the first error you need to call `city.civ` instead of just `civ`

For the second error I also replaced that `uncivInfo` with `city.civ` but you DO NEED to test yourself if it actually works

**Comment by yairm210** [comment]
No, you need to use the *city* to get the uniques, if you want this to be a follower belief
As stated above

**Pull request closed without merging**